tracer/src/Datadog.Trace/Scope.cs

```diff
@@ -33,6 +33,11 @@ internal Scope(Scope parent, Span span, IScopeManager scopeManager, bool finishO
 
         internal Scope Root => Parent?.Root ?? this;
 
+        internal bool FinishOnClose
+        {
+            set => _finishOnClose = value;
+        }
+
```

Comment:
nit: setter-only properties feel weird 😂 An alternative suggestion, add a `Dispose(bool finishOnClose)` overload that does the same thing as 

```csharp
scope.FinishOnClose = false;
scope.Dispose();
```

Reply:
Yeah this does feel weird. The new dispose overload makes sense, but that opens us up to having `Dispose() vs Dispose(bool finishOnClose)` and likely `Close() vs Close(bool finishOnClose)`. I think instead of a set-only, I'll just create a void method instead. Does that sound good to you?